core/kazoo_ledgers-1.0.0/src/kz_ledger.erl

```diff
@@ -70,16 +78,70 @@ debit(Name, Amount, Account, Desc) ->
 %%
 %% @end
 %%--------------------------------------------------------------------
--spec create(ne_binary(), integer(), ne_binary(), api_binary(), ne_binary()) ->
-                    {'ok', ledger()} |
-                    {'error', any()}.
-create(Name, Amount, Account, Desc, Type) ->
+-spec create(ne_binary(), ne_binary(), ne_binary()
+             ,ne_binary(), wh_proplist(), wh_proplist()) -> save_return().
+create(Type, SrcService, SrcId, Account, Usage, Props) ->
     Routines = [
-        fun(L) -> kazoo_ledger:set_name(L, Name) end
-        ,fun(L) -> kazoo_ledger:set_amount(L, Amount) end
-        ,fun(L) -> kazoo_ledger:set_account(L, Account) end
-        ,fun(L) -> kazoo_ledger:set_description(L, Desc) end
-        ,fun(L) -> kazoo_ledger:set_type(L, Type) end
+        fun(L) -> kazoo_ledger:set_type(L, Type) end
```

Comment:
I would like to see this refactored to not use anonymous functions.

```erlang
Routines = [{fun kazoo_ledger:set/type/2, Type}
            ,{fun kazoo_ledger:set_source_service, SrcService}
            ...
           ],
lists:foldl(fun apply_routine/2, kazoo_ledger:new(), Routines).

apply_routine({F, V}, L) -> F(L, V);
apply_routine(F, L) -> F(L).
```

This can be used later in `set_account/2` as well:

```erlang
AccountId = wh_util:format_account_id(Account, 'raw'),
Routines = [{fun kazoo_ledger:set_account_id/2, AccountId}
             ,{fun kazoo_ledger:set_account_name/2, whapps_util:get_account_name(AccountId)}
           ],
lists:foldl(fun apply_routine/2, kazoo_ledger:new(), Routines).
```